substrate/frame/staking/src/migrations.rs

```diff
@@ -91,14 +97,61 @@ pub mod v17 {
 
 			let result = Pallet::<T>::add_electables(prepared_exposures.into_iter());
 			debug_assert!(result.is_ok());
+			log!(info, "v17 applied successfully for pagination, migrated {:?}.", migrated_stashes);
+			// 1x read per history depth and current era read.
+			x = x.saturating_add(T::HistoryDepth::get() + 1u32);
+			// 1x write per exposure migrated.
+			y = y.saturating_add(migrated_stashes);
+
+			let eras_stakers_removed = v16::ErasStakers::<T>::clear(u32::max_value(), None);
+			debug_assert!(eras_stakers_removed.backend == 0);
+			let eras_stakers_clipped_removed =
+				v16::ErasStakersClipped::<T>::clear(u32::max_value(), None);
+			debug_assert!(eras_stakers_clipped_removed.backend == 0);
+			// Just reading and deleting two empty lists
+			x = x.saturating_add(2);
+
+			let mut bounding_storage_success = true;
+
+			x = x.saturating_add(1);
+			let old_invulnerables = v16::Invulnerables::<T>::get();
+			// BoundedVec with MaxInvulnerables limit, this should always succeed
+			let invulnerables_maybe = BoundedVec::try_from(old_invulnerables);
+			match invulnerables_maybe {
+				Ok(invulnerables) => {
+					y = y.saturating_add(1);
+					Invulnerables::<T>::set(invulnerables);
+					log!(info, "v17 applied successfully for Invulnerables.");
+				},
+				Err(_) => {
+					log!(warn, "Migration to v17 failed for Invulnerables.");
+					bounding_storage_success = false;
+				},
+			}
 
-			log!(info, "v17 applied successfully, migrated {:?}.", migrated_stashes);
-			T::DbWeight::get().reads_writes(
-				// 1x read per history depth and current era read.
-				(T::HistoryDepth::get() + 1u32).into(),
-				// 1x write per exposure migrated.
-				migrated_stashes.into(),
-			)
+			x = x.saturating_add(1);
+			let old_disabled_validators = v16::DisabledValidators::<T>::get();
+			// BoundedVec with MaxDisabledValidators limit, this should always succeed
+			let disabled_validators_maybe = BoundedVec::try_from(old_disabled_validators);
+			match disabled_validators_maybe {
+				Ok(disabled_validators) => {
+					y = y.saturating_add(1);
+					DisabledValidators::<T>::set(disabled_validators);
+					log!(info, "v17 applied successfully for Invulnerables.");
+				},
+				Err(_) => {
+					log!(warn, "Migration to v17 failed for DisabledValidators.");
+					bounding_storage_success = false;
+				},
+			}
+
+			if bounding_storage_success {
+				log!(info, "v17 applied successfully to bound storage items.");
+			} else {
+				log!(warn, "v17 failed to bound some storage items.");
+			}
+
+			T::DbWeight::get().reads_writes(x.into(), y.into())
```

Comment:
In any case I will temp remove this migration as we will start from a fresh state and migrate the data. So any failed bounding will be detected in the migration testing.